lib/collection/src/collection/snapshots.rs

```diff
@@ -306,6 +306,12 @@ impl Collection {
         Ok(absolute_snapshot_path)
     }
 
+    pub async fn get_s3_snapshot_path(&self, snapshot_name: &str) -> CollectionResult<PathBuf> {
```

Comment:
This looks like an abstraction leak. Ideally only SnapshotManager should know details of the implementation